docs/content/docs/data-structures/map.md

```diff
@@ -1,10 +1,449 @@
+const pickedUser = users[pickedUserIndex];
+
+// Now store this user object as a whole into the SharedMap
+const task = map.get("task1");
+map.set(task.assignedToKey, pickedUser)
+```
+
+This will work as expected **because the entire object is being stored each time** instead of specific fields.
+
+One way to think about this is that each value stored into the `SharedMap` is a unit of collaboration. Any time you want users to be able to collaborate on invidual pieces of an object, it should be appropriately stored in separate keys.
+
+### Storing shared objects
+
+One of the powerful features of DDSes is that they are infintely nestable. A DDS can be stored in another DDS allowing you to dynamically set up your data hierarchy as best fits your application needs.
+
+This is done using the concept of [handles]({{< relref "data-modeling.md#using-handles-to-store-and-retrieve-fluid-objects" >}}). Handles provide a way for you to serialize your data object into a form that it can be stored into another DDS and loaded back from it.
+
+To do this with `SharedMap`, all you need to do is specify an initial map as part of the `initialObjects` in the `ContainerSchema` and add the `SharedMap` type to `dynamicObjectTypes`.
+
+```javascript
+const schema = {
+    name: "example-container",
+    initialObjects: {
+        initalMap: SharedMap,
+    },
+    dynamicObjectTypes: [SharedMap]
+}
+```
+
+Now, you can dynamically create additional `SharedMap` instances and store their handles into the initial map that is always provided in the container.
+
+```javascript
+const schema = {
+    name: "example-container",
+    initialObjects: {
+        initalMap: SharedMap,
+    },
+    dynamicObjectTypes: [SharedMap]
+}
+
+const { fluidContainer, containerServices } = await client.getContainer(/*service config*/, schema);
+
+const initialMap = fluidContainer.initialObjects.initialMap;
+
+const newSharedMap = fluidContainer.create(SharedMap);
+
+initialMap.set("newSharedMapKey", newSharedMap.handle);
+```
+
+Now, anytime you are looking to load the `newSharedMap` back at a later time, all you have to do is load it back from the stored handle.
+
+```javascript
+const newSharedMapHandle = initialMap.get("newSharedMapKey");
+const newSharedMap = await newSharedMapHandle.get();
+```
+
+{{< callout note >}}
+
+Loading any DDS from its handle is an asynchronous operation. Whereas objects defined in `initialObjects` will be available synchronously once the container is provided, you will need to use an `await` or chain Promises on the `handle.get()` call.
+
+{{< /callout >}}
+
+#### Nested shared objects example
+
+Let's take another look at the example from the **Storing objects** section above to see how it can be updated to use nested `SharedMaps`. This will introduce some data hierarchy and avoid having a flat map that has both the tasks and their fields themselves on the same level. To do so, let's start from the earlier model but with two tasks now.
```

Comment:
```suggestion
You can further extend the example from the [Storing objects]({{< relref "#storing-objects" >}}) section above to see how it can be updated to use nested `SharedMaps`. This will introduce a hierarchy to the data to make it easier to work with. To do so, consider the earlier data model but with two tasks.
```